src/main/java/org/folio/search/service/reindex/ReindexService.java

```diff
@@ -94,20 +96,34 @@ public CompletableFuture<Void> submitFullReindex(String tenantId) {
   }
 
   public CompletableFuture<Void> submitUploadReindex(String tenantId,
-                                                     List<ReindexUploadDto.EntityTypesEnum> entityTypes) {
-    log.info("submitUploadReindex:: for [tenantId: {}, entities: {}]", tenantId, entityTypes);
-    var reindexEntityTypes = entityTypeMapper.convert(entityTypes)
+                                                     ReindexUploadDto reindexUploadDto) {
+    var entityTypes = entityTypeMapper.convert(reindexUploadDto.getEntityTypes())
       .stream().filter(ReindexEntityType::isSupportsUpload).toList();
+    return submitUploadReindex(tenantId, entityTypes, true, reindexUploadDto.getIndexSettings());
+  }
+
+  public CompletableFuture<Void> submitUploadReindex(String tenantId,
+                                                     List<ReindexEntityType> entityTypes) {
+    return submitUploadReindex(tenantId, entityTypes, false, null);
+  }
 
-    validateUploadReindex(tenantId, reindexEntityTypes);
+  private CompletableFuture<Void> submitUploadReindex(String tenantId,
+                                                     List<ReindexEntityType> entityTypes,
+                                                     boolean recreateIndex,
+                                                     IndexSettings indexSettings) {
+    log.info("submitUploadReindex:: for [tenantId: {}, entities: {}]", tenantId, entityTypes);
 
-    for (var reindexEntityType : reindexEntityTypes) {
+    validateUploadReindex(tenantId, entityTypes);
+
+    for (var reindexEntityType : entityTypes) {
       statusService.recreateUploadStatusRecord(reindexEntityType);
-      reindexCommonService.recreateIndex(reindexEntityType, tenantId);
+      if (recreateIndex) {
+        reindexCommonService.recreateIndex(reindexEntityType, tenantId, indexSettings);
```

Comment:
What is the case for not recreating indexes?

Reply:
When it's full reindex. In such case - index settings are passed on merge phase so now indexes are recreated on merge ranges published in order to use passed settings. Upload phase in full flow is triggered on kafka events consuming and so index is not recreated there now